src/AudioUnit/AudioComponent.cs

```diff
@@ -224,6 +395,54 @@ public XamCore.AppKit.NSImage GetIcon ()
 			return new XamCore.AppKit.NSImage (AudioComponentGetIcon (handle));
 		}
 #endif
+		[NoWatch, NoTV, Mac (10,13), iOS (11,0)]
+		[DllImport (Constants.AudioUnitLibrary)]
+		static extern int /* OSStatus */ AudioUnitExtensionSetComponentList (IntPtr /* CFString */ extensionIdentifier, /* CFArrayRef */ IntPtr audioComponentInfo);
+
+		[NoWatch, NoTV, Mac (10,13), iOS (11,0)]
+		[DllImport (Constants.AudioUnitLibrary)]
+		static extern /* CFArrayRef */ IntPtr AudioUnitExtensionCopyComponentList (IntPtr /* CFString */ extensionIdentifier);
+
+		[NoWatch, NoTV, Mac (10,13), iOS (11,0)]
+		public AudioComponentInfo[] ComponentList {
+			get {
+				using (var cfString = new CFString (Name)) {
+					var cHandle = AudioUnitExtensionCopyComponentList (cfString.Handle);
+					if (cHandle == IntPtr.Zero)
+						return null;
+					using (var nsArray = Runtime.GetNSObject<NSArray> (cHandle, owns: true)) {
+						if (nsArray == null)
+							return null;
+						// make things easier for developers since we do not know how to have an implicit conversion from NSObject to AudioComponentInfo
+						var dics = NSArray.FromArray <NSDictionary> (nsArray);
+						var result = new AudioComponentInfo [dics.Length];
+						for (var i = 0; i < result.Length; i++) {
+							result [i] = new AudioComponentInfo (dics[i]);
+						}
+						return result;
+					}
+				}
+			}
+			set {
+				using (var cfString = new CFString (Name)) {
+					var dics = new NSDictionary [value.Length];
```

Comment:
that's a `NRE` if value is `null`
check if `null` is valid (or not) and fix / add validations